Examples/Python/python/acts/examples/detector.py

```diff
@@ -16,420 +17,169 @@
+class CylindricalDetectorContainer:
+    def __init__(
+        self, name, extent, volumes, layers=None, binning=[], loglevel=logging.INFO
+    ):
+        """Create a cylindrical container builder from  volumes or layer definitions
 
-            # Force detector container dimension onto layer
-            layer.setBounds([layer_r_min, layer_r_max, volume_half_z])
-            layer.setTransform(volume_transform)
+        :param name: name of the container
+        :param extent: extent of the container
+        :param volumes: list of volumes
+        :param layers: list of layers [ [extent, provider, binnings, supports], ... ]
+        :param binning: binning of surfces in this container
+        :param loglevel: logging level
 
-            if layer_r_min > last_r_max:
-                gap_bounds = [last_r_max, layer_r_min, volume_half_z]
-                gap_volume = DetectorVolume(
-                    name + "_gap_" + str(igap),
-                    acts.VolumeBoundsType.Cylinder,
-                    gap_bounds,
-                    volume_transform,
-                )
-                complete_volumes += [gap_volume, layer]
-            else:
-                complete_volumes += [layer]
-            # potential last gap
-            if layer_r_max < volume_r_max:
-                gap_bounds = [layer_r_max, volume_r_max, volume_half_z]
-                last_gap = DetectorVolume(
-                    name + "_gap_" + str(igap + 1),
-                    acts.VolumeBoundsType.Cylinder,
-                    gap_bounds,
-                    volume_transform,
-                )
-            else:
-                last_gap = None
-            # Remember the last
-            last_r_max = layer_r_max
-        # The container is z-ordered
-        elif binning is acts.Binning.z:
-            # Force detector container dimension onto layer
-            layer.setBounds(
-                [volume_r_min, volume_r_max, 0.5 * (layer_z_max - layer_z_min)]
-            )
-            if layer_z_min > last_z_max:
-                gap_bounds = [
-                    volume_r_min,
-                    volume_r_max,
-                    0.5 * (layer_z_min - last_z_max),
-                ]
-                gap_transform = acts.Transform3(
-                    [0.0, 0.0, 0.5 * (last_z_max + layer_z_min)]
-                )
-                gap_volume = DetectorVolume(
-                    name + "_gap_" + str(igap),
-                    acts.VolumeBoundsType.Cylinder,
-                    gap_bounds,
-                    gap_transform,
+        """
+        self._name = name
+        self._extent = extent
+        self._layers = layers
+        self._volumes = volumes
+        self._binning = binning
+        self._loglevel = loglevel
+
+    def builder(self):
+        "Return the associated builder"
+        orthogonal = Binning.r
+        if self._binning == Binning.r:
+            orthogonal = Binning.z
```

Comment:
```suggestion
        orthogonal = Binning.r if self._binning == Binning.r else Binning.z
```
But that's a matter of taste, I guess, so I'll leave it up to you

Reply:
Accepted and done.